controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1107,85 +1105,81 @@ def setUp(self, connect):
+        self.assertTrue("msg" in self.context.details)
 
     def test_unpublish_volume_wrong_volume_id(self):
         self.request.volume_id = "some-wrong-id-format"
 
-        context = utils.FakeContext()
-        self.servicer.ControllerUnpublishVolume(self.request, context)
-        self.assertEqual(context.code, grpc.StatusCode.INVALID_ARGUMENT)
+        self.servicer.ControllerUnpublishVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.INVALID_ARGUMENT)
 
     def test_unpublish_volume_wrong_node_id(self):
         self.request.node_id = "some-wrong-id-format"
 
-        context = utils.FakeContext()
-        self.servicer.ControllerUnpublishVolume(self.request, context)
-        self.assertEqual(context.code, grpc.StatusCode.NOT_FOUND)
+        self.servicer.ControllerUnpublishVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.NOT_FOUND)
 
     @patch("controller.controller_server.csi_controller_server.get_agent")
     def test_unpublish_volume_get_host_by_host_identifiers_exception(self, storage_agent):
-        context = utils.FakeContext()
-
         self.mediator.get_host_by_host_identifiers = Mock()
         self.mediator.get_host_by_host_identifiers.side_effect = [array_errors.MultipleHostsFoundError("", "")]
         storage_agent.return_value = self.storage_agent
 
-        self.servicer.ControllerUnpublishVolume(self.request, context)
-        self.assertTrue("Multiple hosts" in context.details)
-        self.assertEqual(context.code, grpc.StatusCode.INTERNAL)
+        self.servicer.ControllerUnpublishVolume(self.request, self.context)
+        self.assertTrue("Multiple hosts" in self.context.details)
+        self.assertEqual(self.context.code, grpc.StatusCode.INTERNAL)
 
         self.mediator.get_host_by_host_identifiers.side_effect = [array_errors.HostNotFoundError("")]
         storage_agent.return_value = self.storage_agent
 
-        self.servicer.ControllerUnpublishVolume(self.request, context)
-        self.assertEqual(context.code, grpc.StatusCode.NOT_FOUND)
+        self.servicer.ControllerUnpublishVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.NOT_FOUND)
 
     @patch("controller.controller_server.csi_controller_server.get_agent")
     def test_unpublish_volume_unmap_volume_excpetions(self, storage_agent):
-        context = utils.FakeContext()
-
-        self.mediator.unmap_volume.side_effect = [array_errors.PermissionDeniedError("msg")]
+        self.mediator.unmap_volume.side_effect = [array_errors.VolumeNotFoundError("vol")]
         storage_agent.return_value = self.storage_agent
-        self.servicer.ControllerUnpublishVolume(self.request, context)
-        self.assertEqual(context.code, grpc.StatusCode.PERMISSION_DENIED)
+        self.servicer.ControllerUnpublishVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)

```

Comment:
why the StatusCode change?

Reply:
I just [changed the order](https://github.com/IBM/ibm-block-csi-driver/pull/209/commits/5e46fb55611cfa737fd21de7f4b060a446225137) of the this function's inner-tests, since it relies on the default status code OK value

Reply:
oh, the mess

Reply:
this mess in the diff is another reason why it's bad practice to have more than 1 "test" per unit test function

Reply:
perhaps it would be easier for you to see the change if you look at the function contents
[before](https://github.com/IBM/ibm-block-csi-driver/blob/81a457c5ca0046ebfaf9f01672176593e51ec51e/controller/tests/controller_server/csi_controller_server_test.py#L1158):
```
def test_unpublish_volume_unmap_volume_excpetions(self, storage_agent):
        context = utils.FakeContext()

        self.mediator.unmap_volume.side_effect = [array_errors.PermissionDeniedError("msg")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, context)
        self.assertEqual(context.code, grpc.StatusCode.PERMISSION_DENIED)

        context = utils.FakeContext()
        self.mediator.unmap_volume.side_effect = [array_errors.HostNotFoundError("host")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, context)
        self.assertEqual(context.code, grpc.StatusCode.NOT_FOUND)

        context = utils.FakeContext()
        self.mediator.unmap_volume.side_effect = [array_errors.UnMappingError("", "", "")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, context)
        self.assertEqual(context.code, grpc.StatusCode.INTERNAL)

        context = utils.FakeContext()
        self.mediator.unmap_volume.side_effect = [array_errors.VolumeNotFoundError("vol")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, context)
        self.assertEqual(context.code, grpc.StatusCode.OK)

        context = utils.FakeContext()
        self.mediator.unmap_volume.side_effect = [array_errors.VolumeAlreadyUnmappedError("")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, context)
        self.assertEqual(context.code, grpc.StatusCode.OK)
```

[after](https://github.com/IBM/ibm-block-csi-driver/blob/5e46fb55611cfa737fd21de7f4b060a446225137/controller/tests/controller_server/csi_controller_server_test.py#L1158):
```
def test_unpublish_volume_unmap_volume_excpetions(self, storage_agent):
        self.mediator.unmap_volume.side_effect = [array_errors.VolumeNotFoundError("vol")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, self.context)
        self.assertEqual(self.context.code, grpc.StatusCode.OK)

        self.mediator.unmap_volume.side_effect = [array_errors.VolumeAlreadyUnmappedError("")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, self.context)
        self.assertEqual(self.context.code, grpc.StatusCode.OK)

        self.mediator.unmap_volume.side_effect = [array_errors.PermissionDeniedError("msg")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, self.context)
        self.assertEqual(self.context.code, grpc.StatusCode.PERMISSION_DENIED)

        self.mediator.unmap_volume.side_effect = [array_errors.HostNotFoundError("host")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, self.context)
        self.assertEqual(self.context.code, grpc.StatusCode.NOT_FOUND)

        self.mediator.unmap_volume.side_effect = [array_errors.UnMappingError("", "", "")]
        storage_agent.return_value = self.storage_agent
        self.servicer.ControllerUnpublishVolume(self.request, self.context)
        self.assertEqual(self.context.code, grpc.StatusCode.INTERNAL)
```